src/services/counters/CounterService.ts

```diff
@@ -0,0 +1,156 @@
+import {App} from "../../ManagementConsole";
+import {DmrService} from "../dmr/DmrService";
+import {ICounter} from "./ICounter";
+import {IDmrRequest} from "../dmr/IDmrRequest";
+import {SocketBindingService} from "../socket-binding/SocketBindingService";
+import IQService = angular.IQService;
+import {LaunchTypeService} from "../launchtype/LaunchTypeService";
+import {isNotNullOrUndefined, traverse, deepValue} from "../../common/utils/Utils";
+import {ICacheContainer} from "../container/ICacheContainer";
+import {CompositeOpBuilder} from "../dmr/CompositeOpBuilder";
+import {IDmrCompositeReq} from "../dmr/IDmrCompositeReq";
+import {JGroupsService} from "../jgroups/JGroupsService";
+import {IServerAddress} from "../server/IServerAddress";
+import {WeakCounter} from "./WeakCounter";
+import {StrongCounter} from "./StrongCounter";
+
+const module: ng.IModule = App.module("managementConsole.services.counter", []);
+
+export class CounterService {
+  static $inject: string[] = ["$q", "jGroupsService", "dmrService", "socketBindingService", "launchType"];
+
+  public static STRONG_COUNTER: string = "STRONG";
+  public static WEAK_COUNTER: string = "WEAK";
+  public static COUNTER_TYPES: string [] = ["STRONG", "WEAK"];
+  public static COUNTER_STORAGE_TYPES: string [] = ["PERSISTENT", "VOLATILE"];
```

Comment:
You've been spending too much time in Java land :smile:. We should export these constants as `const` like we have in other services https://github.com/infinispan/infinispan-management-console/blob/master/src/services/cache-config/CacheConfigService.ts#L21

Reply:
True, good catch